File: android/app/src/main/java/com/albbiz/map/ui/screens/FavoritesScreen.kt
// Bismillah Hir Rahman Nir Raheem
package com.albbiz.map.ui.screens

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.ArrowBack
import androidx.compose.material.icons.filled.Favorite
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.lifecycle.viewmodel.compose.viewModel
import com.albbiz.map.ui.LocalAppStrings
import com.albbiz.map.ui.MeTontGrey
import com.albbiz.map.ui.MeTontRed

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun FavoritesScreen(
    onBackClick: () -> Unit,
    onBusinessClick: (String) -> Unit,
    viewModel: MapViewModel = viewModel()
) {
    val businesses by viewModel.businesses.collectAsState()
    val favoriteIds by viewModel.favoriteIds.collectAsState()
    val userLocation by viewModel.userLocation.collectAsState()
    val strings = LocalAppStrings.current

    val favoriteBusinesses = businesses.filter { it.id in favoriteIds }

    Scaffold(
        topBar = {
            TopAppBar(
                title = {
                    Text(
                        strings.myFavorites,
                        color = Color.White,
                        fontWeight = FontWeight.Bold
                    )
                },
                navigationIcon = {
                    IconButton(onClick = onBackClick) {
                        Icon(
                            Icons.AutoMirrored.Filled.ArrowBack,
                            "Back",
                            tint = Color.White
                        )
                    }
                },
                colors = TopAppBarDefaults.topAppBarColors(
                    containerColor = MeTontRed
                )
            )
        }
    ) { padding ->
        Box(
            modifier = Modifier
                .fillMaxSize()
                //.background(Color(0xFFFFF8F0))
                .background(MaterialTheme.colorScheme.background)
                .padding(padding)
        ) {
            if (favoriteBusinesses.isEmpty()) {
                // ── EMPTY STATE ───────────────────────────────────
                Box(
                    modifier = Modifier.fillMaxSize(),
                    contentAlignment = Alignment.Center
                ) {
                    Column(
                        horizontalAlignment = Alignment.CenterHorizontally,
                        verticalArrangement = Arrangement.spacedBy(12.dp)
                    ) {
                        Surface(
                            modifier = Modifier.size(80.dp),
                            shape = RoundedCornerShape(40.dp),
                            color = MeTontRed.copy(alpha = 0.1f)
                        ) {
                            Box(contentAlignment = Alignment.Center) {
                                Icon(
                                    Icons.Default.Favorite,
                                    null,
                                    modifier = Modifier.size(40.dp),
                                    tint = MeTontRed.copy(alpha = 0.5f)
                                )
                            }
                        }
                        Text(
                            strings.noFavoritesYet,
                            style = MaterialTheme.typography.bodyLarge,
                            color = MeTontGrey,
                            fontWeight = FontWeight.Medium
                        )
                        Text(
                            "Tap the heart icon on any business to save it here",
                            style = MaterialTheme.typography.bodySmall,
                            color = MeTontGrey.copy(alpha = 0.7f)
                        )
                    }
                }
            } else {
                // ── FAVORITES COUNT ───────────────────────────────
                LazyColumn(
                    modifier = Modifier.fillMaxSize(),
                    contentPadding = PaddingValues(16.dp),
                    verticalArrangement = Arrangement.spacedBy(12.dp)
                ) {
                    item {
                        Text(
                            "${favoriteBusinesses.size} saved ${if (favoriteBusinesses.size == 1) "business" else "businesses"}",
                            color = MeTontGrey,
                            fontSize = 13.sp,
                            modifier = Modifier.padding(bottom = 4.dp)
                        )
                    }
                    items(favoriteBusinesses) { business ->
                        BusinessListItemWrapper(
                            business = business,
                            userLocation = userLocation,
                            onClick = { onBusinessClick(business.id) }
                        )
                    }
                }
            }
        }
    }
}

@Composable
fun BusinessListItemWrapper(
    business: com.albbiz.map.data.Business,
    userLocation: com.google.android.gms.maps.model.LatLng?,
    onClick: () -> Unit
) {
    Surface(
        onClick = onClick,
        shape = RoundedCornerShape(16.dp),
        shadowElevation = 2.dp,
        color = Color.White
    ) {
        BusinessListItem(
            business = business,
            userLocation = userLocation,
            isFavorite = true,
            onToggleFavorite = {},
            onClick = onClick
        )
    }
}

File: ios/AlbBizMap_IOS/AlbBizMap/UI/Screens/FavoritesScreen.swift
// Bismillah Hir Rahman Nir Raheem
import SwiftUI

public struct FavoritesScreen: View {
    @Environment(\.appStrings) private var strings

    @ObservedObject public var viewModel: MapViewModel
    public let onBackClick: () -> Void
    public let onBusinessClick: (String) -> Void

    public init(viewModel: MapViewModel, onBackClick: @escaping () -> Void, onBusinessClick: @escaping (String) -> Void) {
        self.viewModel = viewModel
        self.onBackClick = onBackClick
        self.onBusinessClick = onBusinessClick
    }

    public var body: some View {
        VStack(spacing: 0) {
            topAppBar

            let favoriteBusinesses = viewModel.businesses.filter { viewModel.favoriteIds.contains($0.id) }

            if favoriteBusinesses.isEmpty {
                emptyState
            } else {
                ScrollView {
                    VStack(alignment: .leading, spacing: 12) {
                        Text("\(favoriteBusinesses.count) saved \(favoriteBusinesses.count == 1 ? "business" : "businesses")")
                            .font(.system(size: 13))
                            .foregroundColor(.meTontGrey)

                        ForEach(favoriteBusinesses) { biz in
                            BusinessCardView(
                                business: biz,
                                isFavorite: true,
                                onFavoriteToggle: {
                                    viewModel.toggleFavorite(businessId: biz.id, userId: AuthManager.shared.currentUser?.uid ?? "")
                                },
                                onClick: {
                                    onBusinessClick(biz.id)
                                }
                            )
                        }
                    }
                    .padding(16)
                }
            }
        }
        .background(Color.meTontBackground)
    }

    // Matches Android's red TopAppBar (FavoritesScreen.kt).
    private var topAppBar: some View {
        HStack(spacing: 8) {
            Button(action: onBackClick) {
                Image(systemName: "chevron.left")
                    .font(.title3)
                    .foregroundColor(.white)
            }
            Text(strings.myFavorites)
                .font(.system(size: 18, weight: .bold))
                .foregroundColor(.white)
            Spacer()
        }
        .padding(.horizontal, 16)
        .padding(.vertical, 12)
        .background(Color.meTontRed)
    }

    private var emptyState: some View {
        VStack(spacing: 12) {
            Spacer()
            ZStack {
                Circle()
                    .fill(Color.meTontRed.opacity(0.1))
                    .frame(width: 80, height: 80)
                Image(systemName: "heart.fill")
                    .font(.system(size: 32))
                    .foregroundColor(.meTontRed.opacity(0.5))
            }
            Text(strings.noFavoritesYet)
                .font(.system(size: 16, weight: .medium))
                .foregroundColor(.meTontGrey)
            Text("Tap the heart icon on any business to save it here")
                .font(.system(size: 13))
                .foregroundColor(.meTontGrey.opacity(0.7))
                .multilineTextAlignment(.center)
                .padding(.horizontal, 32)
            Spacer()
        }
    }
}
